### PR TITLE
Feature: Notes and Generic API integration in CTIXv3

### PR DESCRIPTION
**Pull request by viktorbab**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @viktorbab and thank you for your contribution! it looks great!
Please see my comments below

File: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py [hunk 1]
```diff
@@ -74,6 +84,29 @@
     "url": "url",
 }
 
+THREAT_DATA_MAP = {
```
Comment (adi88d) [inline Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:87]: remove this const
you don't use it in the code at all

File: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py [hunk 2]
```diff
@@ -1557,20 +1641,253 @@ def file(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
...
+        return results
+
+
+def get_note_details(client: Client, args: Dict[str, Any]) -> CommandResults:
+    id = args["id"]
+    client_url = client.base_url + f"ingestion/notes/{id}/"
+    response = client.get_http_request(client_url)
+
+    if response['status'] == HTTPStatus.BAD_REQUEST:
+        return CommandResults(
```
Comment (adi88d) [inline Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:1689]: maybe its better to return error in this case

File: Packs/CTIX/Integrations/CTIXv3/README.md [hunk 3]
```diff
@@ -1380,4 +1380,173 @@ Lookup to get threat data
               "valid_from": 1652081902,
               "valid_until": null}],
  "total": 1}
-```
\ No newline at end of file
+```
+
+
+
+## ctix-get-all-notes
```
Comment (adi88d) [inline Packs/CTIX/Integrations/CTIXv3/README.md:1387]: please generate the documentation for the new commands using `demisto-sdk generate-docs` command (use the -e flag for using a commad_examples file) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs.

when its done, you can append the docs for the new commands to this file.

File: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py (hunk 1 above)
Comment (viktorbab) [reply Packs/CTIX/Integrations/CTIXv3/CTIXv3.py]: Fixed.

File: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py (hunk 2 above)
Comment (viktorbab) [reply Packs/CTIX/Integrations/CTIXv3/CTIXv3.py]: Fixed.

File: Packs/CTIX/Integrations/CTIXv3/README.md (hunk 3 above)
Comment (viktorbab) [reply Packs/CTIX/Integrations/CTIXv3/README.md:1387]: Should this be taking a long time to complete? When I try running this command, it just hangs endlessly before killing it, even with verbose mode enabled

Comment (viktorbab) [reply Packs/CTIX/Integrations/CTIXv3/README.md]: Fixed.

**Review by adi88d — CHANGES_REQUESTED** [review]
Nice!
please revert the changes in the file `CTIX/Integrations/CTIX/CTIX.yml`

File: Packs/CTIX/Integrations/CTIXv3/README.md [hunk 4]
```diff
@@ -1380,4 +1380,316 @@ Lookup to get threat data
               "valid_from": 1652081902,
               "valid_until": null}],
  "total": 1}
-```
\ No newline at end of file
+```
+
+### domain
```
Comment (adi88d) [inline Packs/CTIX/Integrations/CTIXv3/README.md:1385]: please add documentation for `email` command too and add examples to the commands(use the -e flag for using a commad_examples file) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs.

Do you need any help with something?
Please don’t hesitate to reach out to me here or on Slack

**Comment by edik24** [comment]
Hey @viktorbab, I've sent you a message in the DFIR, can you please reply?

File: Packs/CTIX/Integrations/CTIXv3/README.md (hunk 4 above)
Comment (viktorbab) [reply Packs/CTIX/Integrations/CTIXv3/README.md:1385]: Which email command are you talking about? We don't have an email command.

Comment (adi88d) [reply Packs/CTIX/Integrations/CTIXv3/README.md:1385]: sorry my bad
so just add examples to the command you added to the README(domain, ip, file, url, ctix-get-all-notes, ctix-get-note-details, ctix-create-note, ctix-update-note, ctix-delete-note, ctix-make-request)

Comment (viktorbab) [reply Packs/CTIX/Integrations/CTIXv3/README.md:1385]: Are these manually added? I was under the impression that the `demisto-sdk generate-docs` command should automatically generate the command examples. Is that not correct?

**Pull request merged**